tests/components/time_date/test_sensor.py

```diff
@@ -1,9 +1,19 @@
 """The tests for time_date sensor platform."""
+import pytest
+
 import homeassistant.components.time_date.sensor as time_date
 import homeassistant.util.dt as dt_util
 
 from tests.async_mock import patch
 
+ORIG_TZ = dt_util.DEFAULT_TIME_ZONE
+
+
+@pytest.fixture(autouse=True)
+def restore_ts():
+    """Restore default TZ."""
```

Comment:
Shouldn't this `yield` first?

Reply:
Yes 🤦 

It worked because not all tests override the TZ and so the last one that ran didn't.

Reply:
Thanks to @bdraco for finding the offending commit.